### ISSUE TITLE
Many-to-many relationships pulling too many documents

### ISSUE THREAD

**Issue by faustinoribeiro** [nolanlawson/relational-pouch#70]
First of all, I started testing pouchdb in an ionic 2 proof of concept app only a few days ago, so my knowledge is very limited (and I started to evaluate relational-pouch about 2 hours ago...)

I created the following schema (which, if I am not mistaken, should be a many-to-many relation):
```json
           {
                singular: 'department',
                plural: 'departments',
                relations: {
                    checklists: { hasMany: 'checklist' }
                }
            }, {
                singular: 'checklist',
                plural: 'checklists',
                relations: {
                    departments: { hasMany: 'department' }
                }
            }
```
I created 10 'department' documents of the form { id: 1 ... 10, checklists: [1], ... }
I created 1 checklist:  { Id: 1,  departments: [1,2,3,4,5,6,7,8,9,10], ... }

When running the query db.rel.find('department', [1,3,5,7,9]), all departments (id: 1 to 10) are returned (the single checklist is also returned, which is expected).
This is not the behavior I expected. It seems that because the checklist points to all departments, the departments are pulled along even if they are not in the query list of ids. If I modify the checklist to { id: 1, departments": [1,3,5,7,9] }  to match the query, then it "works" (meaning only departments with ids 1,3,5,7,9 are returned).

Not sure if I am doing something wrong, if it's a bug, or if it's by design. Is there a way to limit documents to what was requested in the list of ids?

**Comment by nolanlawson** [comment]
Yes, you can avoid overfetching by using the `async` attribute: https://github.com/nolanlawson/relational-pouch#async-relationships

Does that answer your question or were you asking about something else?

**Comment by faustinoribeiro** [comment]
Thanks, but this means that I will have to manually side-load the linked documents. Not a big deal, but it kind of defeats the purpose of the library, doesn't it?
As I said in my initial comment, I think it would be more intuitive if automatic inclusion of related documents was restricted to the documents initially queried.
Again not a big deal. Keep the great work!
